src/WinRT.Runtime/ObjectReference.cs

```diff
@@ -231,15 +232,30 @@ public virtual unsafe int TryAs(Guid iid, out IntPtr ppv)
 
         public unsafe IObjectReference As(Guid iid) => As<IUnknownVftbl>(iid);
 
+#if NET
+        [Obsolete(AttributeMessages.GenericDeprecatedMessage)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#endif
         public T AsType<T>()
         {
             ThrowIfDisposed();
+
+#if NET
+            // Same logic as in 'ComWrappersSupport.CreateFactoryForImplementationType', see notes there
+            var attribute = typeof(T).GetCustomAttribute<WinRTImplementationTypeRcwFactoryAttribute>(inherit: false);
```

Comment:
We may still want a non-trim safe fallback as we did in the other PR just in case someone is using it and hadn't moved to new versions of all projections.

Reply:
Done in 63272620e92b9af61fe4b15a5d6bdbeec006c056 👍